### DISCUSSION TITLE
Degradation in release branch support for deployment branch protection rules

### DISCUSSION THREAD

**Discussion by ebronson68** [community/community#64249] [Actions] [answered]
### Select Topic Area

Bug

### Body

As of this week, several of our deployments have failed due to our deployment branch protection rules no longer working. 

We typically create release tags in the `x.x.x` format. Our rule, `*.*.*`, used to work fine. Sometime between 08/16/23 - 08/18/23, our branch protection rules no longer worked and started failing our previously successful deploys.

We've had to disable the deployment branch protection rules entirely for now as there doesn't seem to be a workaround for release branches.

Things I've tried:
- I've even gone back and re-run workflows that successfully deployed using the `*.*.*` rule and the deploy now fails.
- Assuming that there may have been a change in the branch naming pattern syntax, I've tried the following patterns and none of them worked:
  - `*`
  - `**`
  - `**.**.**`
  - `*[0-9].*[0-9]*.[0-9]*
  - `[0-9]*.[0-9]*.[0-9]*`
  - `refs/tags/*`
  - `refs/tags/**`
  - `tags/*`
  - `tags/**`
  - `release/tags/*`
  - `release/tags/**`
- Disabling and re-enabling the deployment branch protection rules
- Tried both pre-release and release events

These are the error messages:
- `Branch "x.x.x" is not allowed to deploy to production due to environment protection rules.`
- `The deployment was rejected or didn't satisfy other protection rules.`

![image](https://github.com/community/community/assets/111298136/8a31addd-4371-4881-b643-180c98e2b751)

Does anyone have any idea as to why this suddenly started happening?

Any help would be greatly appreciated!

**Comment by robertsay** [comment, marked as answer]
Same problem as reported in [this discussion](https://github.com/orgs/community/discussions/62991). Seems to have originated from [this change](https://github.blog/changelog/2023-08-10-actions-runs-triggered-from-tags-or-forks-with-the-same-name-as-a-protected-branch-will-now-be-blocked/).
